examples/rag/README.md

```diff
@@ -0,0 +1,85 @@
+Follow instructions there regarding data preprocessing. A sample finetuning command:
+
+```
+python examples/rag/finetune.py \
+    --data_dir $DATA_DIR \
+    --output_dir $OUTPUT_DIR \
+    --model_name_or_path $MODEL_NAME_OR_PATH \
+    --model_type rag_sequence \
+    --fp16 \
+    --gpus 8
+```
+
+
+# Evaluation
+Apart for parameters specifying the model that's being evaluated and some extra parameters, the evaluation script expects paths to two files:
+- `evaluation_set` - a path file specifying the input dataset for evaluation, a single datapoint per line, e.g.
+```who is the owner of reading football club```
+- `gold_data_path` - a path to a file contaning ground truth answers for samples from the `evaluation_set`.
+
+We expect the following formats of the gold data file:
+
+- for e2e evaluation, we support two formats of gold files:
+    - `qa` - where a single line in the following format: input [tab] output_list, e.g.:
+    ```
+    who is the owner of reading football club	['Xiu Li Dai', 'Dai Yongge', 'Dai Xiuli', 'Yongge Dai']
+    ```
+    - `ans` - where a single line of the gold file contains the expected output string,
+    ```
+    Xiu Li Dai
+    ```
+
+- for retrieval evaluation, we expect a tab-separated list of Wikipedia page titles constituting positive contexts for a given query, e.g. given a question `who sings does he love me with reba`, a line with ground truth retrieval data could look as follows:
+```
+Does He Love You	Does He Love You	Red Sandy Spika dress of Reba McEntire	Greatest Hits Volume Two (Reba McEntire album)	Shoot for the Moon (album)
+```
+
+## Retrieval evaluation
+
+We demonstrate how to evaluate retrieval against DPR evaluation data. You can download respective files from links listed [here](https://github.com/facebookresearch/DPR/blob/master/data/download_data.py#L39-L45).
+
+1. Download and unzip the gold data file. We use the `biencoder-nq-dev` from https://dl.fbaipublicfiles.com/dpr/data/retriever/biencoder-nq-dev.json.gz.
+2. Parse the unziped file using the `parse_dpr_relevance_data.py`
+```
+python examples/rag/parse_dpr_relevance_data.py --src_path path/to/unziped/biencoder-nq-dev.json --evaluation_set path/to/output/biencoder-nq-dev.questions --gold_data_path path/to/output/biencoder-nq-dev.pages
+```
+3. Run evaluation:
+```
+python examples/rag/eval_rag.py \
+    --model_name_or_path $MODEL_NAME_OR_PATH \ # model name or path of the model we're evaluating
+    --model_type rag_sequence \ # RAG model type (rag_token or rag_sequence)
+    --evaluation_set path/to/output/biencoder-nq-dev.questions \ # an input dataset for evaluation
+    --gold_data_path path/to/output/biencoder-nq-dev.pages \ # a dataset containing ground truth answers for samples from the evaluation_set
+    --predictions_filename retrieval_preds.tsv  \ # name of file in which predictions will be stored
```

Comment:
In L68(here), L70, and L81, `predictions_filename` should change to `predictions_path`.

Reply:
It is not related to this issue, but is it expected behavior that the code currently does not work with $MODEL_NAME_OR_PATH specified as `facebook/rag-sequence-nq`? I checked that the model is uploaded to https://huggingface.co/models...

Reply:
Hi, thanks for the comments!
- re: `predictions_filename` - `predictions_path` - will fix this
- the script should work with `facebook/rag-sequence-nq` - will double check, there has been some back and forth between calling the tokenizer directory `question_encoder_tokenizer` vs `question_encoder` (and similarly for the `generator`) so there may be some discrepancy leftover between the directory name on s3 and what is expected in `from_pretrained`
- the index we're releasing here is a quantized version of a DPR index we used to obtain results from the RAG paper - our initial index was much larger and post factum we managed to quantize it and significantly reduce the size without hurting the downstream performance.

Reply:
I see, thank you a lot for the detailed and informative response!
- Thank you for double-checking the automated download by `from_pretrained`.
- I see, that is why the paper does not mention SQ :) Could you let me know when do you schedule to publicize `hf_bert_base.hnswSQ8_correct_phi_128.c_index`, or is there a temporary link to download the index at this time?

Reply:
Hi @soheeyang - feel free to ping me an email if you want to discuss the quantization details more - the short story, as Ola mentions, is that the old index was very RAM intensive, so to make RAG work on more people's machines, we compressed it, whilst preserving as much accuracy as possible.